antarest/study/model.py

```diff
@@ -160,6 +161,111 @@ def __repr__(self) -> str:  # pragma: no cover
         return f"{cls_name}({study_id=}, {tag=})"
 
 
+class DirectoryGroup(Base):
+    """
+    A table to manage the many-to-many relationship between `Directory` and `Group`
+
+    Attributes:
+        directory_id: The ID of the directory associated with the group.
+        group_id: The ID of the group associated with the directory.
+    """
+
+    __tablename__ = "directory_group"
+    __table_args__ = (PrimaryKeyConstraint("directory_id", "group_id"),)
+
+    directory_id: Mapped[str] = mapped_column(
+        String(36), ForeignKey("directory.id", ondelete="CASCADE"), index=True, nullable=False
+    )
+    group_id: Mapped[str] = mapped_column(
+        String(36), ForeignKey("groups.id", ondelete="CASCADE"), index=True, nullable=False
+    )
+
+    @override
+    def __str__(self) -> str:  # pragma: no cover
+        cls_name = self.__class__.__name__
+        return f"[{cls_name}] directory_id={self.directory_id}, group={self.group_id}"
+
+    @override
+    def __repr__(self) -> str:  # pragma: no cover
+        cls_name = self.__class__.__name__
+        directory_id = self.directory_id
+        group_id = self.group_id
+        return f"{cls_name}({directory_id=}, {group_id=})"
+
+
+class Directory(Base):
+    """
+    Represents a logical directory for organizing studies in the managed workspace.
+
+    Directories are stored in the database and provide a hierarchical organization
+    for studies, independent of the physical filesystem structure.
+
+    Attributes:
+        id: The unique identifier of the directory (UUID).
+        name: The non-qualified name of the directory (e.g., "project1").
+        parent_id: The ID of the parent directory, or None for root directories.
+        owner_id: The ID of the owner of the directory.
+    """
+
+    __tablename__ = "directory"
+
+    id: Mapped[str] = mapped_column(
+        String(36),
+        primary_key=True,
+        default=lambda: str(uuid.uuid4()),
+        unique=True,
+    )
+    name: Mapped[str] = mapped_column(String(255), nullable=False)
+    parent_id: Mapped[Optional[str]] = mapped_column(
+        String(36), ForeignKey("directory.id", name="fk_directory_parent_id"), nullable=True, index=True
+    )
+    owner_id: Mapped[Optional[int]] = mapped_column(Integer, ForeignKey(Identity.id), nullable=True, index=True)
+
```

Comment:
I think we will also need a "public mode", to keep the wording we use for studies, similar to the permissions for "others" under linux.

Otherwise, we will not be able to have directories visible by everyone.